### PR TITLE
driver deployment logging

### PR DESCRIPTION
**Pull request by pohly**
This is in response to https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/view/pmem-csi/job/pmem-csi/job/devel/187 which got stuck without useful information:

```
May 18 15:39:19.061: INFO: wait for PMEM-CSI: services "pmem-csi-metrics" not found
```

was the last message and then it just gets stuck?

### CODE REVIEW

File: test/start-operator.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ echo "Deploying '${deploy}'..."
 
 if [ -f "$deploy" ]; then
   tmpdir=$(${SSH} mktemp -d)
-  trap '${SSH} "echo cleaning up temp folder: $tmpdir; rm -rf $tmpdir"' SIGTERM SIGINT EXIT
+  trap '${SSH} "rm -rf $tmpdir"' SIGTERM SIGINT EXIT
```
Comment (pohly) [inline test/start-operator.sh:21]: @avalluri this debug output is no longer needed, right?

Comment (avalluri) [reply test/start-operator.sh:21]: Yes, it's ok to remove.

**Comment by pohly** [comment]
The logging change itself seems to be okay (but of course doesn't fix the underlying issue):
- with it: once per minute "INFO: Still waiting for PMEM-CSI driver, last error: get controller metrics: ..." in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/view/pmem-csi/job/pmem-csi/view/change-requests/job/PR-647/3/execution/node/123/log/?consoleFull
- without it: much more frequent "INFO: wait for PMEM-CSI: get controller metrics:" in https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/view/pmem-csi/job/pmem-csi/view/change-requests/job/PR-642/5/consoleFull

**Comment by pohly** [comment]
@avalluri: I had to rebase and add two more changes:
- 78eee57
- 8353266 

Please have another look.

**Comment by pohly** [comment]
Do not merge. Fedora 1.15 doesn't come up anymore: https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/view/pmem-csi/job/pmem-csi/job/devel-next2/3

**Comment by pohly** [comment]
Fedora 1.15 didn't have the `EndpointSlices` feature gate yet. Fixed with this conditional check:
```
diff --git a/test/test-config.sh b/test/test-config.sh
index 5d7e9378..b40e4f02 100644
--- a/test/test-config.sh
+++ b/test/test-config.sh
@@ -88,10 +88,6 @@ fi
 # (https://github.com/kubernetes/kubernetes/issues/70082).
 : ${TEST_ETCD_VOLUME:=}
 
-# Kubernetes feature gates to enable/disable.
-# EndpointSlice is disabled because of https://github.com/kubernetes/kubernetes/issues/91287
-: ${TEST_FEATURE_GATES:=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true,EndpointSlice=false}
-
 # Device mode that test/setup-deployment.sh is using.
 # Allowed values: lvm, direct
 # This string is used as part of deployment file name.
@@ -141,6 +137,11 @@ fi
 # to use the Kubernetes version that ships with it.
 : ${TEST_KUBERNETES_VERSION:=1.18}
 
+# Kubernetes feature gates to enable/disable.
+# EndpointSlice is disabled because of https://github.com/kubernetes/kubernetes/issues/91287
+: ${TEST_FEATURE_GATES:=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true\
+$(case ${TEST_KUBERNETES_VERSION} in 1.1[0-5]) ;; *) echo ',EndpointSlice=false';; esac)}
+
 # If non-empty, the version of Kata Containers which is to be installed
 # in the Kubernetes cluster. Installation is done with
 # https://hub.docker.com/r/katadocker/kata-deploy
```

**Comment by pohly** [comment]
`ImagePullBackOff - Back-off pulling image "quay.io/k8scsi/csi-node-driver-registrar:v1.2.0"` caused a test failure because the driver did not come up.

**Comment by pohly** [comment]
@avalluri: one more commit for the image pull problem - is 
c19025a okay?

**Comment by okartau** [comment]
One failure in the latest PR run
https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/job/pmem-csi/view/change-requests/job/PR-647/14/testReport/junit/fedora-1_18.imagefile/ext4/512Mi/
shows similar signs as I describe in #654, (but not exactly same), seems something goes wrong in retrieving of fstype.

**Comment by pohly** [comment]
> seems something goes wrong in retrieving of fstype

Should be fixed now. I had missed some direct ssh calls (i.e. those without the retry loop) and after replacing those, printing the command in the helper function got mixed into the output.

**Comment by avalluri** [comment]
> @avalluri: one more commit for the image pull problem - is
> [c19025a](https://github.com/intel/pmem-csi/commit/c19025a1896ab8d04f9718f3f1273d344e27bd54) okay?

Yes, it's good that you provided a variable to configure that.

**Comment by pohly** [comment]
Finally! Clean full test run: https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/job/pmem-csi/job/devel-next2/9

@avalluri: is c30c6ca  okay? Then let's merge.

**Comment by avalluri** [comment]
> @avalluri: is [c30c6ca](https://github.com/intel/pmem-csi/commit/c30c6ca4665563f58dbc73e707abd01fc19fa124) okay? 

Yes, let's merge this PR.

**Pull request merged**